README.md
To package Nginx, clone this repo, clean it...
```
find debian/ -name '\.git*' -exec rm -rf {} \; &>/dev/null
rm -r debian/modules/nginx-upload-progress/test/
rm -r debian/modules/ruby-passenger/debian/
rm -r debian/modules/ruby-passenger/rpm/
rm -r debian/modules/ruby-passenger/test/
rm -r debian/modules/nginx-rtmp-module/test/
rm README.md
```
... and build it your favorite way


debian/changelog
nginx (1.14.2-1~dotdeb+8.1) oldoldstable; urgency=medium

  [ Guillaume Plessis ]
  * New upstream version 1.14.2

 -- Guillaume Plessis <gui@dotdeb.org>  Tue, 04 Dec 2018 20:03:48 +0000

nginx (1.14.1-1~dotdeb+8.1) oldoldstable; urgency=medium

  [ Guillaume Plessis ]
  * New upstream version 1.14.1
  * Update submodules
  * Patches refresh for 1.14.1

 -- Guillaume Plessis <gui@dotdeb.org>  Thu, 08 Nov 2018 03:02:42 +0000

nginx (1.12.2-1~dotdeb+8.1) oldstable; urgency=medium

  [ Guillaume Plessis ]
  * New upstream version 1.12.2

 -- Guillaume Plessis <gui@dotdeb.org>  Thu, 19 Oct 2017 23:11:03 +0000

nginx (1.12.1-1~dotdeb+8.1) oldstable; urgency=high

  [ Guillaume Plessis ]
  * Imported Upstream version 1.12.1

 -- Guillaume Plessis <gui@dotdeb.org>  Thu, 20 Jul 2017 13:34:25 +0000

nginx (1.12.0-1~dotdeb+8.1) oldstable; urgency=medium

  [ Guillaume Plessis ]
  * Imported Upstream version 1.12.0
  * Update submodules
  * Patches refresh for 1.12.0
  * nginx-x-rid-header is obsolete
  * updates configure options

 -- Guillaume Plessis <gui@dotdeb.org>  Mon, 12 Jun 2017 21:30:15 +0000

nginx (1.10.3-1~dotdeb+8.1) stable; urgency=medium

  [ Guillaume Plessis ]
  * Updates nginx-auth-ldap
  * Remove Pinba from nginx-full and nginx-extras
  * Imported Upstream version 1.10.3

 -- Guillaume Plessis <gui@dotdeb.org>  Wed, 01 Feb 2017 15:50:19 +0000

nginx (1.10.2-2~dotdeb+8.2) stable; urgency=medium

  * Fix a Security Regression

 -- Guillaume Plessis <gui@dotdeb.org>  Fri, 28 Oct 2016 05:39:42 +0000

nginx (1.10.2-2~dotdeb+8.1) stable; urgency=medium

  [ Christos Trochalakis ]
  * debian/nginx-common.postinst:
    + Secure log file handling (owner & permissions) against privilege
      escalation attacks. /var/log/nginx is now owned by root:adm.
      Thanks Dawid Golunski (http://legalhackers.com) for the report.
      Changing /var/log/nginx permissions effectively reopens #701112,
      since log files can be world-readable. This is a trade-off until
      a better log opening solution is implemented upstream (trac:376).
  * debian/control:
    Don't allow building against liblua5.1-0-dev on architectures
    that libluajit is available. (Closes: #826167)

 -- Guillaume Plessis <gui@dotdeb.org>  Thu, 27 Oct 2016 18:17:42 +0000

nginx (1.10.2-1~dotdeb+8.1) stable; urgency=medium

  * Imported Upstream version 1.10.2

 -- Guillaume Plessis <gui@dotdeb.org>  Tue, 18 Oct 2016 23:34:27 +0000

nginx (1.10.1-1~dotdeb+8.2) stable; urgency=medium

  * Generate dbgsym packages only if dh_strip supports it
  * naxsi is back!
  * nginx-upload-progress now loads perfectly

 -- Guillaume Plessis <gui@dotdeb.org>  Fri, 03 Jun 2016 23:58:31 +0000

nginx (1.10.1-1~dotdeb+8.1) stable; urgency=medium

  * Imported Upstream version 1.10.1
  * ngx_pagespeed 1.11.33.2

 -- Guillaume Plessis <gui@dotdeb.org>  Wed, 01 Jun 2016 02:55:05 +0000

nginx (1.10.0-1~dotdeb+8.1) stable; urgency=medium

  * Imported Upstream version 1.10.0
  * ngx_pagespeed 1.11.33.1

 -- Guillaume Plessis <gui@dotdeb.org>  Sat, 07 May 2016 17:28:42 -0800

nginx (1.8.1-1~dotdeb+8.1) stable; urgency=medium

  * Imported Upstream version 1.8.1
  * ngx_pagespeed 1.10.33.2

 -- Guillaume Plessis <gui@dotdeb.org>  Tue, 26 Jan 2016 13:00:50 -0800

nginx (1.8.0-1~dotdeb+8.2) stable; urgency=medium

  * nginx-push-stream 0.5.1. Close #62
  * ngx-conf in /usr/sbin. Close #66
  * ngx_pagespeed 1.9.32.4. Fixes #68

 -- Guillaume Plessis <gui@dotdeb.org>  Wed, 17 Jun 2015 14:50:19 -0700

nginx (1.8.0-1~dotdeb+8.1) oldstable; urgency=medium

  * Imported Upstream version 1.8.0
  * syslog support is now upstream
  * Thread pool support
  * Sync with Debian

 -- Guillaume Plessis <gui@dotdeb.org>  Sat, 25 Apr 2015 20:39:38 -0700

nginx (1.6.3-1~dotdeb+wheezy.1) stable; urgency=medium

  * Modules update
  * Imported Upstream version 1.6.3
  * generic Dockerfile to ease Debian packaging
  * gbp.conf

 -- Guillaume Plessis <gui@dotdeb.org>  Wed, 08 Apr 2015 11:47:13 -0700

nginx (1.6.2-1~dotdeb.1) stable; urgency=medium

  * Imported Upstream version 1.6.2
  * Patches refresh for 1.6.2
  * ngx_pagespeed 1.9.32.1
  * Modules update
  * nginx-rtmp-module added to nginx-extras

 -- Guillaume Plessis <gui@dotdeb.org>  Tue, 16 Sep 2014 23:11:44 -0700

nginx (1.6.1-1~dotdeb.1) stable; urgency=medium

  * New upstream release
  * Sync with Debian
  * Modules update

 -- Guillaume Plessis <gui@dotdeb.org>  Thu, 14 Aug 2014 15:54:42 +0200

nginx (1.6.0-1~dotdeb.2) stable; urgency=medium

  * ngx_pagespeed security upgrade

 -- Guillaume Plessis <gui@dotdeb.org>  Fri, 27 Jun 2014 07:32:42 -0700

nginx (1.6.0-1~dotdeb.1) stable; urgency=medium

  * Rebuilt for Dotdeb

 -- Guillaume Plessis <gui@dotdeb.org>  Fri, 09 May 2014 15:23:42 -0700

nginx (1.6.0-1) unstable; urgency=medium

  [ Christos Trochalakis ]
  * New upstream release.
  * debian/rules:
    + Enable auth request http module. (Closes: #725732)
  * debian/patches/series:
    + Remove backported openssl guard patch.

 -- Christos Trochalakis <yatiohi@ideopolis.gr>  Thu, 24 Apr 2014 19:23:22 +0300

nginx (1.4.7-2) unstable; urgency=medium

  [ Christos Trochalakis ]
  * debian/rules:
    + Convert to dh $@
      As a side effect, fixes make dependency issues (Closes: #744050)

 -- Christos Trochalakis <yatiohi@ideopolis.gr>  Thu, 17 Apr 2014 14:30:27 +0300

nginx (1.4.7-1) unstable; urgency=medium

  [ Christos Trochalakis ]
  * New upstream release. (Closes: #742059)
    + Fixes pottential arbitrary code execution (CVE-2014-0133)
      Debian build was not vulnerable since it was compiled
      with the --with-debug configure option.
  * debian/modules/nginx-http-push:
    + Update to v0.711 (Closes: #732251)
  * debian/modules/headers-more-nginx-module:
    + Update to v0.25, containing several bugfixes.
  * debian/modules/nginx-echo:
    + Update to v0.51, containing several bugfixes.
  * debian/modules/nginx-dav-ext-module:
    + Update to v0.0.3, containing some uri encoding fixes.

src/http/modules/perl/Makefile.PL

# Copyright (C) Igor Sysoev
# Copyright (C) Nginx, Inc.

use 5.006001;
use Config;
use ExtUtils::MakeMaker;

WriteMakefile(
    NAME              => 'nginx',
    VERSION_FROM      => 'nginx.pm',     # finds $VERSION
    PREREQ_PM         => {},             # e.g., Module::Name => 1.1

    ABSTRACT_FROM     => 'nginx.pm',     # retrieve abstract from module
    AUTHOR            => 'Igor Sysoev',

    CCFLAGS           => "$ENV{NGX_PM_CFLAGS}",
    # Pass link hardening flags
    # $Config{lddlflags} is the default
    LDDLFLAGS         => "$Config{lddlflags} $ENV{DEBIAN_NGINX_PERL_LDFLAGS}",
    OPTIMIZE          => '-O',

    INC               => join(" ", map {
                             m#^/# ? "-I $_" : "-I ../../../../../$_"
                         } (split /\s+/, $ENV{NGX_INCS})),

    depend => {
        'nginx.c'     => join(" ", map {
                             m#^/# ? $_ : "../../../../../$_"
                         } (split(/\s+/, $ENV{NGX_DEPS}),
                            "src/http/modules/perl/ngx_http_perl_module.h"))
    },

    PM => {
        'nginx.pm'    => '$(INST_LIBDIR)/nginx.pm'
    }
);


debian/modules/naxsi/naxsi-ui/SQLWrapper.py
#!/usr/bin/env python

from ConfigParser import ConfigParser
import os

class SQLWrapperException(Exception):
    pass

class SQLWrapper(object):    
    def __init__(self, config_file):
        try:
            fd = open(config_file)
        except IOError:
            raise SQLWrapperException('Cannot open config file %s' % config_file)
        self.conf = ConfigParser()
        self.conf.readfp(fd)
        self.dbtype = self.get_config('dbtype')
        if self.dbtype == 'sqlite':
            self.DBManager = __import__('sqlite3')
        elif self.dbtype == 'mysql':
            self.DBManager = __import__('MySQLdb')
        else:
            raise SQLWrapperException('Unhandled db type : %s' % self.get_config('dbtype'))
        if self.dbtype == 'mysql':
            self.user = self.get_config('username')
            self.host = self.get_config('hostname')
            self.password = self.get_config('password')
        self.dbname = self.get_config('dbname')

    def get_config(self, key):
        return self.conf.get('sql', key)

    def connect(self):
        if self.dbtype == 'mysql':
            self.__conn = self.DBManager.connect(self.host, self.user, self.password, self.dbname)
            self.__cursor = self.__conn.cursor(self.DBManager.cursors.DictCursor)
        else:
            self.__conn = self.DBManager.connect(self.dbname)
            self.__conn.row_factory = self.DBManager.Row
            self.__conn.text_factory = str  # to avoid problems with encoding
            self.__cursor = self.__conn.cursor()

    def setRowToDict(self):
        if self.dbtype == 'sqlite3':
            self.__conn.row_factory = self.DBManager.Row
        elif self.dbtype == 'mysql':
            self.__cursor = self.__conn.cursor(self.DBManager.cursors.DictCursor)

    def execute(self, query, args = None):
        if args is None:
            self.__cursor.execute(query)
        else:
            if self.dbtype == 'sqlite':
                query = query.replace('%s', '?') #hmmmm....
            self.__cursor.execute(query, args)
        # if self.dbtype == 'sqlite':
        #     self.__conn.commit()

    def StartInsert(self):
        if self.dbtype == 'sqlite':
            self.__conn.execute("BEGIN")

        pass
    def StopInsert(self):
        if self.dbtype == 'sqlite':
            self.__conn.commit()
        elif self.dbtype == 'mysql':
            self.__conn.commit()

    def getResults(self):
        return self.__cursor.fetchall()


    def getLastId(self):
        return self.__cursor.lastrowid


    def drop_database(self):
        if self.dbtype == 'mysql':
            self.__cursor.execute("DROP DATABASE IF EXISTS %s;" % self.dbname)
        elif self.dbtype == 'sqlite':
            os.unlink(os.getcwd() + '/' + self.dbname)

    def create_db(self):
        if self.dbtype == 'mysql':
            self.__cursor.execute("CREATE DATABASE %s;" %  self.dbname)
        elif self.dbtype == 'sqlite':
            self.connect()
        

    def select_db(self, dbname):
        if self.dbtype == 'mysql':
            self.__conn.select_db(dbname)


    def create_all_tables(self):
        if self.dbtype == 'mysql':
            self.execute("CREATE TABLE connections (url_id INTEGER auto_increment, id_exception INTEGER , date TIMESTAMP default CURRENT_TIMESTAMP, peer_ip TEXT, host TEXT, primary key(url_id, id_exception))")
            self.execute("CREATE TABLE urls (url_id INTEGER auto_increment primary key, url TEXT)")
            self.execute("CREATE TABLE exceptions (exception_id INTEGER auto_increment, zone TEXT, var_name TEXT, rule_id INTEGER , primary key (exception_id, rule_id))")            
        elif self.dbtype == 'sqlite':
            self.execute("CREATE TABLE connections (url_id INTEGER, id_exception INTEGER, date TIMESTAMP default CURRENT_TIMESTAMP, peer_ip TEXT, host TEXT, primary key(url_id, id_exception))")
            self.execute("CREATE TABLE urls (url_id INTEGER primary key, url TEXT)")
            self.execute("CREATE TABLE exceptions (exception_id INTEGER PRIMARY KEY AUTOINCREMENT, zone TEXT, var_name TEXT, rule_id INTEGER)")


    def getWhitelist(self):
        self.execute('select e.exception_id as id, e.zone as zone, e.var_name as var_name, e.rule_id as rule_id, u.url as url from exceptions as e join connections as c on (c.id_exception = e.exception_id) join urls as u on (c.url_id = u.url_id)')
        return self.getResults()


debian/modules/nginx-development-kit/src/hash/murmurhash2.c


#ifndef MURMURHASH2_C
#define MURMURHASH2_C

#define     MURMURHASH2_DIGEST_LENGTH   4

//-----------------------------------------------------------------------------
// MurmurHash2, by Austin Appleby

// Note - This code makes a few assumptions about how your machine behaves -

// 1. We can read a 4-byte value from any address without crashing
// 2. sizeof(int) == 4

// And it has a few limitations -

// 1. It will not work incrementally.
// 2. It will not produce the same results on little-endian and big-endian
//    machines.

unsigned int MurmurHash2 ( const void * key, int len, unsigned int seed )
{
	// 'm' and 'r' are mixing constants generated offline.
	// They're not really 'magic', they just happen to work well.

	const unsigned int m = 0x5bd1e995;
	const int r = 24;

	// Initialize the hash to a 'random' value

	unsigned int h = seed ^ len;

	// Mix 4 bytes at a time into the hash

	const unsigned char * data = (const unsigned char *)key;

	while(len >= 4)
	{
		unsigned int k = *(unsigned int *)data;

		k *= m; 
		k ^= k >> r; 
		k *= m; 
		
		h *= m; 
		h ^= k;

		data += 4;
		len -= 4;
	}
	
	// Handle the last few bytes of the input array

	switch(len)
	{
	case 3: h ^= data[2] << 16;
	case 2: h ^= data[1] << 8;
	case 1: h ^= data[0];
	        h *= m;
	};

	// Do a few final mixes of the hash to ensure the last few
	// bytes are well-incorporated.

	h ^= h >> 13;
	h *= m;
	h ^= h >> 15;

	return h;
}

#endif


debian/modules/nginx-pinba/pinba.pb-c.c
/* Generated by the protocol buffer compiler.  DO NOT EDIT! */

#include "pinba.pb-c.h"
void   pinba__request__init
                     (Pinba__Request         *message)
{
  static Pinba__Request init_value = PINBA__REQUEST__INIT;
  *message = init_value;
}
size_t pinba__request__get_packed_size
                     (const Pinba__Request *message)
{
  PROTOBUF_C_ASSERT (message->base.descriptor == &pinba__request__descriptor);
  return protobuf_c_message_get_packed_size ((const ProtobufCMessage*)(message));
}
size_t pinba__request__pack
                     (const Pinba__Request *message,
                      uint8_t       *out)
{
  PROTOBUF_C_ASSERT (message->base.descriptor == &pinba__request__descriptor);
  return protobuf_c_message_pack ((const ProtobufCMessage*)message, out);
}
size_t pinba__request__pack_to_buffer
                     (const Pinba__Request *message,
                      ProtobufCBuffer *buffer)
{
  PROTOBUF_C_ASSERT (message->base.descriptor == &pinba__request__descriptor);
  return protobuf_c_message_pack_to_buffer ((const ProtobufCMessage*)message, buffer);
}
Pinba__Request *
       pinba__request__unpack
                     (ProtobufCAllocator  *allocator,
                      size_t               len,
                      const uint8_t       *data)
{
  return (Pinba__Request *)
     protobuf_c_message_unpack (&pinba__request__descriptor,
                                allocator, len, data);
}
void   pinba__request__free_unpacked
                     (Pinba__Request *message,
                      ProtobufCAllocator *allocator)
{
  PROTOBUF_C_ASSERT (message->base.descriptor == &pinba__request__descriptor);
  protobuf_c_message_free_unpacked ((ProtobufCMessage*)message, allocator);
}
static const ProtobufCFieldDescriptor pinba__request__field_descriptors[19] =
{
  {
    "hostname",
    1,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_STRING,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, hostname),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "server_name",
    2,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_STRING,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, server_name),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "script_name",
    3,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_STRING,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, script_name),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "request_count",
    4,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_UINT32,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, request_count),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "document_size",
    5,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_UINT32,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, document_size),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "memory_peak",
    6,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_UINT32,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, memory_peak),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "request_time",
    7,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_FLOAT,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, request_time),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "ru_utime",
    8,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_FLOAT,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, ru_utime),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "ru_stime",
    9,
    PROTOBUF_C_LABEL_REQUIRED,
    PROTOBUF_C_TYPE_FLOAT,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, ru_stime),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "timer_hit_count",
    10,
    PROTOBUF_C_LABEL_REPEATED,
    PROTOBUF_C_TYPE_UINT32,
    PROTOBUF_C_OFFSETOF(Pinba__Request, n_timer_hit_count),
    PROTOBUF_C_OFFSETOF(Pinba__Request, timer_hit_count),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "timer_value",
    11,
    PROTOBUF_C_LABEL_REPEATED,
    PROTOBUF_C_TYPE_FLOAT,
    PROTOBUF_C_OFFSETOF(Pinba__Request, n_timer_value),
    PROTOBUF_C_OFFSETOF(Pinba__Request, timer_value),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "timer_tag_count",
    12,
    PROTOBUF_C_LABEL_REPEATED,
    PROTOBUF_C_TYPE_UINT32,
    PROTOBUF_C_OFFSETOF(Pinba__Request, n_timer_tag_count),
    PROTOBUF_C_OFFSETOF(Pinba__Request, timer_tag_count),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "timer_tag_name",
    13,
    PROTOBUF_C_LABEL_REPEATED,
    PROTOBUF_C_TYPE_UINT32,
    PROTOBUF_C_OFFSETOF(Pinba__Request, n_timer_tag_name),
    PROTOBUF_C_OFFSETOF(Pinba__Request, timer_tag_name),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "timer_tag_value",
    14,
    PROTOBUF_C_LABEL_REPEATED,
    PROTOBUF_C_TYPE_UINT32,
    PROTOBUF_C_OFFSETOF(Pinba__Request, n_timer_tag_value),
    PROTOBUF_C_OFFSETOF(Pinba__Request, timer_tag_value),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "dictionary",
    15,
    PROTOBUF_C_LABEL_REPEATED,
    PROTOBUF_C_TYPE_STRING,
    PROTOBUF_C_OFFSETOF(Pinba__Request, n_dictionary),
    PROTOBUF_C_OFFSETOF(Pinba__Request, dictionary),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "status",
    16,
    PROTOBUF_C_LABEL_OPTIONAL,
    PROTOBUF_C_TYPE_UINT32,
    PROTOBUF_C_OFFSETOF(Pinba__Request, has_status),
    PROTOBUF_C_OFFSETOF(Pinba__Request, status),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "memory_footprint",
    17,
    PROTOBUF_C_LABEL_OPTIONAL,
    PROTOBUF_C_TYPE_UINT32,
    PROTOBUF_C_OFFSETOF(Pinba__Request, has_memory_footprint),
    PROTOBUF_C_OFFSETOF(Pinba__Request, memory_footprint),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "requests",
    18,
    PROTOBUF_C_LABEL_REPEATED,
    PROTOBUF_C_TYPE_MESSAGE,
    PROTOBUF_C_OFFSETOF(Pinba__Request, n_requests),
    PROTOBUF_C_OFFSETOF(Pinba__Request, requests),
    &pinba__request__descriptor,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
  {
    "schema",
    19,
    PROTOBUF_C_LABEL_OPTIONAL,
    PROTOBUF_C_TYPE_STRING,
    0,   /* quantifier_offset */
    PROTOBUF_C_OFFSETOF(Pinba__Request, schema),
    NULL,
    NULL,
    0,            /* packed */
    0,NULL,NULL    /* reserved1,reserved2, etc */
  },
};
static const unsigned pinba__request__field_indices_by_name[] = {
  14,   /* field[14] = dictionary */
  4,   /* field[4] = document_size */
  0,   /* field[0] = hostname */
  16,   /* field[16] = memory_footprint */
  5,   /* field[5] = memory_peak */
  3,   /* field[3] = request_count */
  6,   /* field[6] = request_time */
  17,   /* field[17] = requests */
  8,   /* field[8] = ru_stime */
  7,   /* field[7] = ru_utime */
  18,   /* field[18] = schema */
  2,   /* field[2] = script_name */
  1,   /* field[1] = server_name */
  15,   /* field[15] = status */
  9,   /* field[9] = timer_hit_count */
  11,   /* field[11] = timer_tag_count */
  12,   /* field[12] = timer_tag_name */
  13,   /* field[13] = timer_tag_value */
  10,   /* field[10] = timer_value */
};
static const ProtobufCIntRange pinba__request__number_ranges[1 + 1] =
{
  { 1, 0 },
  { 0, 19 }
};
const ProtobufCMessageDescriptor pinba__request__descriptor =
{
  PROTOBUF_C_MESSAGE_DESCRIPTOR_MAGIC,
  "Pinba.Request",
  "Request",
  "Pinba__Request",
  "Pinba",
  sizeof(Pinba__Request),
  19,
  pinba__request__field_descriptors,
  pinba__request__field_indices_by_name,
  1,  pinba__request__number_ranges,
  (ProtobufCMessageInit) pinba__request__init,
  NULL,NULL,NULL    /* reserved[123] */
};
